docs: rework readme (#1017)
faker-js/faker: CONTRIBUTING.md

The sources are located in the [src](src) directory.
All fake data generators are divided into namespaces (each namespace being a separate module).
Most of the generators use the _definitions_, which are just plain JavaScript objects/arrays/strings that are separate for each [locale](src/locales).

## Building Faker

The project is being built by [esbuild](https://esbuild.github.io) (see [bundle.ts](scripts/bundle.ts))

```shell
pnpm install
pnpm run build
```

## Testing

> ```shell

Reviewer: Maybe some words what framework we use for testing (like line 26) or why we think tests are important.

Reviewer 2: For me it would be more important to mention that you always have to run build when changing the code, as tests are run from built code, not from source, which was quite unusual for me...

I would not mention `install` as this seems to be quite obvious step.

pnpm install
pnpm run build

pnpm run test
# or
pnpm run coverage
```

You can view a code coverage report generated in `coverage/index.html`.

## Developing the docs

```shell

> # build the Faker dist

Reviewer 2: This should be mention as a text, and not inside the code example I think...

# it's used inside of certain routes
pnpm run build

pnpm run docs:dev
```

> ## Building and serving the docs statically

Reviewer 2: It would be nice to explain why this is needed and how this differ from the docs:dev

```shell
# build the Faker dist
# it's used inside of certain routes
pnpm run build

pnpm run docs:build # Output docs to /dist
pnpm run docs:serve # Serve docs from /dist
```

> ## Deploying Documentation

Reviewer 2: ```suggestion
## Deploying documentation
```

> The website is kindly hosted for free by the Netlify team under their Open Source plan. See the [netlify.toml](netlify.toml) for configuration.

Reviewer 2: This is already mentioned on the main Readme so is this worth the repetition? It's not connected with contribution to the project, right?

**localisation**

I would add new section documenting how to add a new locale. People ask about it. See https://discord.com/channels/929487054990110771/929544823109742593/981252262515060817

The sources are located in the [src](src) directory.
All fake data generators are divided into namespaces (each namespace being a separate module).
Most of the generators use the _definitions_, which are just plain JavaScript objects/arrays/strings that are separate for each [locale](src/locales).

## Building Faker

The project is being built by [esbuild](https://esbuild.github.io) (see [bundle.ts](scripts/bundle.ts))

```shell
pnpm install
pnpm run build
```

## Testing

> Before you can run the tests, you need to install all dependencies and also need to build the project, because some tests depend on the bundled content.

Reviewer 3: ```suggestion
Before you can run the tests, you need to install all dependencies and build the project, because some tests depend on the bundled content.
```

```shell
pnpm install
pnpm run build

pnpm run test
# or
pnpm run coverage
```

> You can view a code coverage report generated in `coverage/index.html`.

Reviewer 3: ```suggestion
You can view a generated code coverage report at `coverage/index.html`.
```

## Adding new locale or updating existing one

> After you have updated the locale definitions, you need to run `pnpm run generate:locales` to generate/update the related files.

Author: @ST-DDT please check this section and provide suggestion or even update this yourself as you have access to the branch :slightly_smiling_face:

> After you have updated the locale definitions, you need to run `pnpm run generate:locales` to generate/update the related files.

Reviewer 4: ```suggestion
After adding new or updating existing locale data, you need to run `pnpm run generate:locales` to generate/update the related files.
```
